File: core/core-enum/src/main/java/com/imstargg/core/enums/DateRange.java
package com.imstargg.core.enums;

import java.time.LocalDate;
import java.util.List;
import java.util.stream.Stream;

public enum DateRange {

    ONE_DAY(1),
    THREE_DAYS(3),
    ONE_WEEK(7),
    TWO_WEEKS(14),
    ;

    private final int value;

    DateRange(int value) {
        this.value = value;
    }

    List<LocalDate> lastDates(LocalDate date) {
        return Stream.iterate(date, d -> d.minusDays(1))
                .limit(value)
                .toList();
    }
}

File: imstargg-frontend/src/model/enums/DateRange.ts
export const DateRangeValue = {
  THREE_DAYS: 'THREE_DAYS',
  ONE_WEEK: 'ONE_WEEK',
  TWO_WEEKS: 'TWO_WEEKS',
  ONE_MONTH: 'ONE_MONTH',
} as const;

export type DateRange = typeof DateRangeValue[keyof typeof DateRangeValue];

export const DateRangeValues = Object.keys(DateRangeValue) as DateRange[];

export const dateRangeTitle = (dateRange: DateRange): string => {
  switch (dateRange) {
    case DateRangeValue.THREE_DAYS:
      return '3일';
    case DateRangeValue.ONE_WEEK:
      return '1주';
    case DateRangeValue.TWO_WEEKS:
      return '2주';
    case DateRangeValue.ONE_MONTH:
      return '1개월';
  }
}
